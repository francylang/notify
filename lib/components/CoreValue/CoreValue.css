.character-card {
  background-color: #1a90bf;
  opacity: 0.5;
  align-items: center;
  text-align: center;
  width: 137px;
  position: relative;
  display: flex;
  justify-content: center;
  flex-direction: column;
  margin: 10px 10px 40px 28px;
  padding: 38px 50px;
  box-shadow: 2px 4px 12px #888888;
  border-radius: 8%;
  cursor: pointer; }

.no-opacity {
  background-color: #1a90bf;
  opacity: 1.5;
  align-items: center;
  text-align: center;
  width: 137px;
  position: relative;
  display: flex;
  justify-content: center;
  flex-direction: column;
  margin: 10px 10px 40px 28px;
  padding: 38px 50px;
  box-shadow: 2px 4px 12px #888888;
  border-radius: 8%;
  cursor: pointer; }

.character-card:hover {
  opacity: 1.5; }

.core-value {
  text-decoration: none;
  font-size: 28px;
  color: #fff;
  cursor: pointer; }

.core-value-counter {
  font-size: 80px;
  color: #fff;
  cursor: pointer; }

.green {
  background-color: #688737; }

.red {
  background-color: #d35656; }

.blue {
  background-color: #3364FF; }

.purple {
  background-color: #A46CC3; }

.yellow {
  background-color: #ECAF4B; }

.navy {
  background-color: #405780; }

.orange {
  background-color: #FF9633; }

@media screen and (max-width: 500px) {
  .character-card {
    margin: 29px 8px 40px 7px; }
  .no-opacity {
    width: 70%;
    cursor: pointer; } }
