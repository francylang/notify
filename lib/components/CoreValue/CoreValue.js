import React from 'react';
import PropTypes from 'prop-types';

const CoreValue = ({ coreValue, plusOneCoreValue, studentId, count, color }) => {

  return (
    <div
      className={count ? `no-opacity ${color}` : `character-card ${color}`}
      onClick={() => plusOneCoreValue(coreValue, studentId)}>
        <article className='card'>
          <h3 className='core-value'>{ coreValue }</h3>
            <h4 className='core-value-counter'>
                { count }
            </h4>
        </article>
    </div>
  );
};

CoreValue.propTypes = {
  coreValue: PropTypes.string,
  plusOneCoreValue: PropTypes.func,
  studentId: PropTypes.number,
  count: PropTypes.number,
  color: PropTypes.string
};

export default CoreValue;
