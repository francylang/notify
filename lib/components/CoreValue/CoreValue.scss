.character-card {
  background-color: #49A6C3;
  align-items: center;
  text-align: center;
  width: 10%;
  position: relative;
  display: flex;
  justify-content: center;
  flex-direction: column;
  margin: 10px;
  margin-bottom: 40px;
  padding: 65px;
  box-shadow: 2px 4px 12px #888888;
  border-radius: 20%;
  cursor: pointer;
}

.core-value {
  text-decoration: none;
  font-size: 28px;
  color: #fff;
  cursor: pointer;

}

.core-value-counter {
  font-size: 28px;
  color: #fff;
  padding-top: 28px;
  cursor: pointer;
}

.green {
  background-color: #688737;
}

.red {
  background-color: #49A6C3;
}

.blue {
  background-color: #405780;
}

.purple {
  background-color: #A46CC3;
}

.yellow {
  background-color: #ECAF4B;
}
