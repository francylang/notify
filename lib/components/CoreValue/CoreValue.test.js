import React from 'react';
import CoreValue from './CoreValue.js';
import { shallow } from 'enzyme';

describe('<CoreValue />', () => {

  it('should exist', () => {
    const wrapper = shallow(<CoreValue />);

    expect(wrapper).toBeDefined();
  });

  it('should match the CoreValue snapshot', () => {
    const wrapper = shallow(<CoreValue />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render a core value card', () => {
    const wrapper = shallow(<CoreValue />);

    const card = wrapper.find('.card');

    expect(card.length).toEqual(1);
  });
});
