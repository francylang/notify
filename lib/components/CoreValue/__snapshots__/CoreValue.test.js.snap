// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CoreValue /> should match the CoreValue snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="character-card undefined"
    onClick={[Function]}
>
    <article
        className="card"
    >
        <h3
            className="core-value"
        />
        <h4
            className="core-value-counter"
        />
    </article>
</div>,
  "nodes": Array [
    <div
      className="character-card undefined"
      onClick={[Function]}
>
      <article
            className="card"
      >
            <h3
                  className="core-value"
            />
            <h4
                  className="core-value-counter"
            />
      </article>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <CoreValue />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="character-card undefined"
          onClick={[Function]}
>
          <article
                    className="card"
          >
                    <h3
                              className="core-value"
                    />
                    <h4
                              className="core-value-counter"
                    />
          </article>
</div>,
        "_debugID": 4,
        "_renderedOutput": <div
          className="character-card undefined"
          onClick={[Function]}
>
          <article
                    className="card"
          >
                    <h3
                              className="core-value"
                    />
                    <h4
                              className="core-value-counter"
                    />
          </article>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CoreValue />,
}
`;
